src/main/java/com/google/sps/servlets/PromotionQuizServlet.java

```diff
@@ -21,36 +26,44 @@
   private static final String JSON_CONTENT_TYPE = "application/json";
   private static final String QUIZ_SUBMIT = "promotionQuizSubmit";
   // this will be replaced by the game stage id queried from the user
-  private static final String HARDCODED_QUIZ_KEY = "softwareEngineerLevel1";
   private static final String PROMOTED_MESSAGE =
       "Congratulations, you passed the quiz and were promoted!";
   private static final String NOT_PROMOTED_MESSAGE =
       "You did not pass the quiz. Study the content and try again later";
-  private static final Double CORRECT_NEEDED_THRESHOLD = 0.5;
-
-  private static final Gson gson = new Gson();
+  private static final String IS_FINAL_STAGE_MESSAGE =
+      "Congratulations! You reached the final stage! You may no longer be promoted in this path.";
```

Comment:
This is a bit of a weird message. What is the user supposed to do? Start over in another path? Stop playing? We should provide messaging that makes it clear what they should do next,

Reply:
I think this message should almost never be displayed in normal game flow, only if the user manages to get to promotionquiz.html on their own. However, you make a good point about providing a next step. If a future PR, I will be handling how all these messages will be displayed in the front end.